### ISSUE TITLE
Crash in Android Oreo

### ISSUE THREAD

**Issue by idragon81** [razorpay/razorpay-android-sample-app#41]
When Running the app in Android Oreo the app crashes with the following stack trace

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ft.gt.assistant/com.razorpay.CheckoutActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.graphics.Bitmap.compress(android.graphics.Bitmap$CompressFormat, int, java.io.OutputStream)' on a null object reference
                                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
                                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
                                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
                                                                       at android.os.Handler.dispatchMessage(Handler.java:105)
                                                                       at android.os.Looper.loop(Looper.java:164)
                                                                       at android.app.ActivityThread.main(ActivityThread.java:6541)
                                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
                                                                    Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.graphics.Bitmap.compress(android.graphics.Bitmap$CompressFormat, int, java.io.OutputStream)' on a null object reference
                                                                       at com.razorpay.B_$q$.d__1_(:14114)
                                                                       at com.razorpay.CheckoutActivity.onCreate(:23479)
                                                                       at android.app.Activity.performCreate(Activity.java:6975)
                                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1213)
                                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
                                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892) 
                                                                       at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593) 
                                                                       at android.os.Handler.dispatchMessage(Handler.java:105) 
                                                                       at android.os.Looper.loop(Looper.java:164) 
                                                                       at android.app.ActivityThread.main(ActivityThread.java:6541) 
                                                                       at java.lang.reflect.Method.invoke(Native Method) 
                                                                       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240) 
                                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

When running on a Nougat Device or emulator, there is no crash, however when running in an Android Oreo Emulator, the issue can be seen

I am using 1.4.0 Version of the library

**Comment by idragon81** [comment]
Looks like the issue is because of the adaptive icons introduced in Oreo and because I was using the ic_launcher mipmap.

You should either handle it by properly parsing the background and foreground of the adaptive icon or you should check if the resource type is actually a drawable and check if the drawable is not null before calling the compress method

**Comment by hariramsvnit** [comment]
@idragon81 Thanks for reporting it, we will fix it soon.

**Comment by hariramsvnit** [comment]
@idragon81 Not able to reproduce the crash, can you help me with that?
I have tried setting the image by `setImage(R.mipmap.ic_launcher)`, but it's not crashing on Android 8, I have tried it on sample app(https://github.com/razorpay/razorpay-android-sample-app).

**Comment by hariramsvnit** [comment]
fixed in version `1.4.4`
